controllers/controllers.go

```diff
@@ -177,8 +169,20 @@ func windowsNodePredicate(byoh bool) predicate.Funcs {
 			return false
 		},
 		DeleteFunc: func(e event.DeleteEvent) bool {
-			return false
+			return isByohWindowsNode(e.Object, byoh)
 		},
 	}
 
 }
+
+// isByohWindowsNode returns true if the node object has the Windows and BYOH labels present
+func isByohWindowsNode(o client.Object, byoh bool) bool {
```

Comment:
I am not sure that this method is named/purposed correctly. This method is checking if it is either a Windows BYOH node (with BYOH label) or a Windows Machine node (without BYOH label), depending on the 2nd parameter.